pennylane_lightning/core/src/simulators/lightning_kokkos/gates/BasicGeneratorFunctors.hpp

```diff
@@ -371,86 +355,249 @@ void applyGenMultiRZ(
         });
 }
 
+template <class PrecisionT, class FuncT> class applyNCGenerator1Functor {
+    using KokkosIntVector = Kokkos::View<std::size_t *>;
+
+    Kokkos::View<Kokkos::complex<PrecisionT> *> arr;
+    const FuncT core_function;
+    KokkosIntVector indices;
+    KokkosIntVector parity;
+    KokkosIntVector rev_wires;
+    KokkosIntVector rev_wire_shifts;
+    std::size_t mask{0U};
+    static constexpr std::size_t one{1U};
+    std::size_t i0;
+    std::size_t i1;
+
+  public:
+    template <class ExecutionSpace>
+    applyNCGenerator1Functor([[maybe_unused]] ExecutionSpace exec,
+                             Kokkos::View<Kokkos::complex<PrecisionT> *> arr_,
+                             std::size_t num_qubits,
+                             const std::vector<std::size_t> &controlled_wires,
+                             const std::vector<bool> &controlled_values,
+                             const std::vector<std::size_t> &wires,
+                             FuncT core_function_)
+        : arr(arr_), core_function(core_function_) {
+        const std::size_t n_contr = controlled_wires.size();
+        const std::size_t n_wires = wires.size();
+        const std::size_t nw_tot = n_contr + n_wires;
+        PL_ASSERT(n_wires == 1);
```

Comment:
Feel free to ignore it, but I think we can do something like:
```suggestion
        const std::size_t n_wires = wires.size();
        const std::size_t nw_tot = controlled_wires.size() + n_wires; // n_contr + n_wires
        PL_ASSERT(n_wires == 1);
```

Reply:
I've updated it, thanks

Reply:
I've changed a few lines below which now uses `n_contr`

Reply:
Nice!